arch/3ds/Makefile.in

```diff
@@ -62,6 +62,10 @@ arch/3ds/%.o: arch/3ds/%.c arch/3ds/shader_2d.shbin.o arch/3ds/shader_playfield.
 	$(if ${V},,@echo "  CC      " $<)
 	${CC} -MD ${core_cflags} ${core_flags} -Wno-unused-macros -c $< -o $@
 
+arch/3ds/%.o: arch/3ds/%.cpp
+	$(if ${V},,@echo "  CXX     " $<)
+	${CXX} -MD ${core_cxxflags} ${core_flags} -Wno-unused-macros -c $< -o $@
+
```

Comment:
The shbin.o dependencies need to be moved from the .c rule to the .cpp rule. Parallel builds rely on that to complete properly.